config/config.go

```diff
@@ -463,6 +473,17 @@ func (c *ScrapeConfig) MarshalYAML() (interface{}, error) {
 	return discovery.MarshalYAMLWithInlineConfigs(c)
 }
 
+// StorageConfig configures runtime reloadable configuration options.
+type StorageConfig struct {
```

Comment:
Are we missing some custom `UnmarshalYAML` functions like for other structs in the file to attach the default config in case no config was provided?

Reply:
I wasn't sure how the default config and `UnmarshalYAML` should work honestly. Does insertion of default config show up in the config display within Prometheus' UI? It could be confusing if we inserted default exemplar config when the user hasn't enabled the experimental feature itself. 

The only other bit of default config I kind of understood was the `QueueConfig`, which is inserted when we have a remote write that doesn't have a user defined queue config. In the case of exemplar config, we're not also requiring the user to enable the exemplar storage via the config file, so how do we decide to insert the default config?

Reply:
```go
	// Ugly hack to get the previous default size.
	if opts.MaxExemplars == 0 {
		opts.MaxExemplars = config.DefaultExemplarsConfig.MaxExemplars
	}
```

Regarding this code, I feel having 0 in the config is a legit option? In that case this could be a surprise. The earlier behaviour of having it 0 when set 0 sounds better.

And your point is also valid. So my suggestion here is to make it a pointer, like `ExemplarsConfig *ExemplarsConfig `yaml:"exemplars,omitempty"``, and in `main.go` we assign the default if it is `nil` (and not for 0)